src/python/gudhi/representations/vector_methods.py

```diff
@@ -350,6 +351,92 @@ def __call__(self, diag):
         """
         return self.fit_transform([diag])[0,:]
 
+
+class IrregularBettiCurve(BaseEstimator, TransformerMixin):
+    """
+    This is a class for computing Betti curves from a list of persistence diagrams. Unlike the BettiCurve class, the curves are not sampled at fixed, regular points. Each curve has all its data intact.
+    """
+
+    def __init__(self, filtration_range = [np.nan, np.nan]):
+        """
+        Constructor for the IrregularBettiCurve class.
+
+        Parameters:
+            filtration_range ([float, float]): minimum and maximum filtration range to consider, of the form [x_min, x_max] (default [numpy.nan, numpy.nan]). The behavior is not the same as for sample_range in BettiCurve. In particular, intervals born before x_min are not considered. Intervals born in [x_min, x_max) contribute to the Betti curves as if they were infinite.
+        """
+        self.filtration_range = np.array(filtration_range)
+
+    def fit(self, X, y = None):
+        """
+        See the corresponding function in BettiCurve.
+        """
+        if np.isnan(self.filtration_range).any():
+            pre = DiagramScaler(use=True, scalers=[([0], MinMaxScaler()), ([1], MinMaxScaler())]).fit(X,y)
+            [mx,my],[Mx,My] = [pre.scalers[0][1].data_min_[0], pre.scalers[1][1].data_min_[0]], [pre.scalers[0][1].data_max_[0], pre.scalers[1][1].data_max_[0]]
+            self.filtration_range = np.where(np.isnan(self.filtration_range), np.array([mx, My]), np.array(self.filtration_range))
+
+        return self
+
+    def transform(self, X):
+        """
+        Compute the Betti curve for each persistence diagram.
+
+        Parameters:
+            X (list of n x 2 NumPy arrays): input persistence diagrams.
+        
+        Returns:
+            Pair (xs, bettis), where xs is a length-k NumPy array with all filtration values for which any of the Betti curves change value, and bettis is an (number of diagrams) x k NumPy integer array with the values of the k Betti curves at the corresponding filtration values. The value bettis[i, k] is the value of the Betti curve of the k'th diagram on the interval [xs[i], xs[i+1]).
```

Comment:
The doc is a bit confusing to me: didn't you mean "bettis is an (number of diagrams) x k NumPy integer array with the values of the **(number of diagrams)** Betti curves at the corresponding filtration values." and "The value bettis[i, k] is the value of the Betti curve of the **i**'th diagram on the interval [xs[**k**], xs[**k+1**])."?

Reply:
I most certainly did. Thanks.